### PR TITLE
Facts style conformance and backporting

### PR DESCRIPTION
**Pull request by gguillotte**
Followup PR to #679 to quote keys and backport edits to the rest of Puppet 4.

CC @nfagerlund @jtappa @ahpook

### CODE REVIEW

File: source/puppet/4.1/reference/lang_facts_and_builtin_vars.markdown [hunk 1]
```diff
@@ -79,11 +78,11 @@ All facts appear in Puppet as [top-scope variables][topscope]. They can be acces
 
 Example, with the osfamily fact:
 
-~~~ ruby
-    if $osfamily == 'redhat' {
-      # ...
-    }
-~~~
+``` puppet
+if $osfamily == 'redhat' {
```
Comment (ahpook) [inline source/puppet/4.1/reference/lang_facts_and_builtin_vars.markdown]: this one should also be `$facts['os']['family']`

File: source/puppet/4.1/reference/lang_facts_and_builtin_vars.markdown [hunk 2]
```diff
@@ -79,11 +78,11 @@ All facts appear in Puppet as [top-scope variables][topscope]. They can be acces
 
 Example, with the osfamily fact:
```
Comment (ahpook) [inline source/puppet/4.1/reference/lang_facts_and_builtin_vars.markdown]: maybe: "Example, using the `family` fact, which aggregates all Red Hat-derived Linux variants into one 'redhat' value."

**Comment by ahpook** [comment]
found one `$osfamily` that wasn't updated, other than that it looks great. Thanks @gguillotte

File: source/puppet/4.1/reference/lang_facts_and_builtin_vars.markdown (hunk 1 above)
Comment (gguillotte) [inline source/puppet/4.1/reference/lang_facts_and_builtin_vars.markdown]: The context here is to explicitly refer to this as a "classic `$fact_name` fact" distinct from the following section, which covers "the `$facts['fact_name']` hash". I might be missing something here, but I'm not sure that replacing this specific $fact_name example with a $facts['fact_name'] example makes sense.

Comment (ahpook) [inline source/puppet/4.1/reference/lang_facts_and_builtin_vars.markdown]: ah, doh, that makes sense.  missed it from the line diffs. Ignore me :)

**Comment by gguillotte** [comment]
@jtappa I think that's a 👍  from @ahpook. Not sure if you want to take a look before this gets merged.

**Comment by jtappa** [comment]
:+1:

**Pull request merged**